locator/tablets.cc

```diff
@@ -670,9 +670,23 @@ future<bool> check_tablet_replica_shards(const tablet_metadata& tm, host_id this
 }
 
 class tablet_effective_replication_map : public effective_replication_map {
+
+public:
+    using replication_factor_index_t = uint16_t;
+    using replication_factor_list = utils::small_vector<replication_factor_index_t, 8>;
```

Comment:
The element of replication_factor_list is replication factor, so the type shouldn't be `replication_factor_index_t` (dc index), it's not an index into the list, it's replication factor value.